File: JavaScript/PollingSystem/Poll.js
class Poll {
  question = '';
  options = {};
  scheduledTime = Date();
  validTill = Date();
  createBy = 0;

  constructor(question, options, createdBy) {
    if (!question || question == '' || !options || options.length <= 1 || !createdBy) {
      throw new Error('Invalid poll parameters!');
    }
    this.question = question;
    this.options = options;
    this.createBy = createdBy;
  }
}

module.exports = {Poll};

File: Go/PollingSystem-go/poll.go
package main

import (
	"fmt"
	"time"
)

type Poll struct {
	ID            int
	Question      string
	Options       []string
	ScheduledTime time.Time
	ValidTill     time.Time
	CreateBy      int
}

func NewPoll(question string, options []string, createdBy int) (*Poll, error) {
	if question == "" || len(options) <= 1 || createdBy == 0 {
		return nil, fmt.Errorf("invalid poll parameters!")
	}
	return &Poll{
		Question:      question,
		Options:       options,
		ScheduledTime: time.Now(),
		ValidTill:     time.Time{},
		CreateBy:      createdBy,
	}, nil
}
